Tools/Scripts/collect-pgo-profiles

```diff
@@ -0,0 +1,92 @@
+    ensure_empty_directory(benchmark_profile_directory)
+    with tempfile.TemporaryDirectory() as temp_dir:
+        output_file = os.path.join(temp_dir, f'{benchmark}.json')
+        command = [sys.executable, harness, '--plan', benchmark, '--diagnose-directory', benchmark_profile_directory,
+                   '--generate-pgo-profiles', '--http-server-type', 'builtin', '--count', '1',
+                   '--output-file', output_file, *run_benchmark_args]
+        if verbose:
+            command.append('--debug')
+        logger.info(f'Running {benchmark} with {shlex.join(command)}')
+        subprocess.run(command, check=True)
+
+    merge_command = [sys.executable, PGO_PROFILE, 'merge', benchmark_profile_directory]
+    if verbose:
+        merge_command.append('--verbose')
+    subprocess.run(merge_command, check=True)
+
+    return benchmark_profile_directory
+
+
+def main():
+    parser = argparse.ArgumentParser(prog='collect-pgo-profiles',
+                                     allow_abbrev=False,
+                                     description='PGO profile collection script, '
+                                                 'unknown arguments will be passed to run-benchmark harness.')
+    parser.add_argument('-r', '--run-benchmark-harness', required=False, default=DEFAULT_RUN_BENCHMARK,
+                        help=f'Location for run-benchmark harness, defaults to {DEFAULT_RUN_BENCHMARK}')
+    parser.add_argument('-b', '--benchmarks', nargs='+', required=True,
+                        help='Benchmarks to run for PGO profile collection.')
+    parser.add_argument('-o', '--output-directory', required=True,
+                        help='PGO profile output directory.')
+    parser.add_argument('-p', '--compressed-profile-sub-path', required=True,
+                        help='Sub path under output directory for compressed PGO profiles.')
+    parser.add_argument('-v', '--verbose', action='store_true', default=False,
+                        help='Turn on debug logging')
+    args, run_benchmark_args = parser.parse_known_args()
+
+    if args.verbose:
+        logger.setLevel(logging.DEBUG)
+
+    ensure_empty_directory(args.output_directory)
+
+    combine_command = [sys.executable, PGO_PROFILE, 'combine']
+    for benchmark in args.benchmarks:
+        profile_directory = collect_pgo_profiles(args.run_benchmark_harness, benchmark, args.output_directory,
+                                                 run_benchmark_args, args.verbose)
+        combine_command.extend([f'--{benchmark}', profile_directory])
+
+    combined_profile_directory = os.path.join(args.output_directory, 'output')
+    ensure_empty_directory(combined_profile_directory)
+    combine_command.extend(['--output', combined_profile_directory])
+    subprocess.run(combine_command, check=True)
+
+    compressed_profile_directory = os.path.join(args.output_directory, args.compressed_profile_sub_path)
+    ensure_empty_directory(compressed_profile_directory)
```

Comment:
Is there any reason not to just use `os.makedir` here? Given we started with an empty directory, it would be very surprising if it existed.